awscli/s3transfer/copies.py

```diff
@@ -300,6 +323,52 @@ def _get_transfer_size(
             return total_transfer_size - (part_index * part_size)
         return part_size
 
+    def _check_destination_object_existence(self, client, call_args):
+        """
+        Check if destination object exists when IfNoneMatch is specified.
+
+        This method implements the ifNoneMatch header functionality by checking if the
+        destination object already exists before proceeding with the copy operation.
+        Args:
```

Comment:
nit: if documenting args, then we should use existing standards for docstrings, which is to use Sphinx markup. See here for an example for parameters and return values.

https://github.com/aws/aws-cli/blob/e0dba6bc2aa29ab31302419d6764fcd5fb069f96/awscli/argprocess.py#L138-L145